File: CSharp.Functions/Common/Entities/FirmDetails/EmployersLiability.cs
namespace Common.Entities
{
    public class EmployersLiability : InsurerBrokerInformation
    {
        public FirmBasicInfo Insurer { get; set; } = new();
        public FirmBasicInfo Broker { get; set; } = new();
        public Money PremiumAmount { get; set; } = new();
        public Money SingleIndemnityLimit { get; set; } = new();
        public Money AggregateIndemnityLimit { get; set; } = new();
        public Money PolicyExcess { get; set; } = new();
        public List<FileModel> ProposalFormFile { get; set; } = new();
        public List<FileModel> SchedOfInsurance { get; set; } = new();
        public string? PaymentFrequency { get; set; }
        public string? PolicyExclusions { get; set; }
        public long? RetroactiveStartDate { get; set; }
        public long? StartDate { get; set; }
        public long? EndDate { get; set; }
    }
}

File: vue.jsonforms/src/entities/employers-liability/EmployersLiability.ts
import { FirmBasicInfo } from "@/entities/FirmBasicInfo";
import { Money } from "@/entities/Money";
import { ContactNumber } from "../ContactNumber";
import {FileEntity} from "@/entities/FileEntity";

export class EmployersLiability {
  public paymentFrequency?: string;
  public retroactiveStartDate?: number;
  public startDate?: number;
  public endDate?: number;
  public policyExclusions?: string;
  public insurer: FirmBasicInfo = new FirmBasicInfo();
  public broker: FirmBasicInfo = new FirmBasicInfo();
  public premiumAmount: Money = new Money();
  public singleIndemnityLimit: Money = new Money();
  public aggregateIndemnityLimit: Money = new Money();
  public policyExcess: Money = new Money();
  public proposalFormFile: FileEntity[] = [];
  public schedOfInsurance: FileEntity[] = [];
  public insurerRegisteredAddress?: string;
  public insurerTradingAddress?: string;
  public insurerIsTradingSameAsRegisteredAddress: boolean = false;
  public insurerEmailAddress?: string;
  public insurerContactNumber?: ContactNumber;
  public insurerWebsite?: string;
  public isBrokerResponsible: boolean | null = null;
  public isBrokerCompany: boolean | null = true;
  public brokerRegisteredAddress: string | undefined;
  public brokerTradingAddress: string | undefined;
  public brokerIsTradingSameAsRegisteredAddress: boolean = false;
  public brokerEmailAddress?: string;
  public brokerContactNumber?: ContactNumber;
  public brokerWebsite?: string;
}